server/src/db/migrations/20211102194503_machines_table.ts

```diff
@@ -0,0 +1,50 @@
+import { Knex } from "knex";
+
+
+export async function up(knex: Knex): Promise<void> {
+    
+    knex.schema.hasTable('MachineFamilies').then(function(exists) {
+        if (!exists) {
+            return knex.schema.createTable('MachineFamilies', function(t) {
+                t.increments('id');
+                t.string('name', 50);
+                t.text('description');
+                t.integer('training_module');
+                t.foreign('training_module').references('id').inTable('TrainingModule');
+              });
+        }
+    });
+
+    knex.schema.hasTable('Machines').then(function(exists) {
+        if (!exists) {
+          return knex.schema.createTable('Machines', function(t) {
+            t.increments('id');
+            t.string('name', 50);
+            t.integer('category');
+            t.foreign('category').references('id').inTable('MachineFamilies');
+            t.string('room', 5);
+            t.date('date_added').defaultTo(knex.fn.now());
+            t.boolean('reserved').defaultTo(false);
+            t.uuid('reserved_by');
```

Comment:
Would it makes sense to have a separate table for reservations? This implementation would only allow the current reservation to be known, while a reservation table could have past and future reservations for each machine 

for example: 
| reservation_id | machine_id | student_id | starttime | endtime |